restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/fcm/FcmConfigurationClient.java

```diff
@@ -508,34 +486,50 @@ private FcmBinding createOrUpdateBinding(FcmBinding binding, String bindingSid){
     }
 
 
-    public boolean deleteBinding(String bindingSid){
+    public void deleteBinding(String bindingSid) throws RCException{
         RCLogger.v(TAG, "deleteBinding method started");
-        boolean result = false;
         HttpURLConnection connection = null;
         try{
             String url = "https://" + pushDomain + "/" + this.pushPath + "/bindings/" + bindingSid;
             RCLogger.v(TAG, "calling url: " + url);
             connection = createUrlRequestWithUrl(url);
             connection.setRequestMethod("DELETE");
-            int responseCode = connection.getResponseCode();
 
-            if (responseCode == HttpURLConnection.HTTP_OK){
-               result = true;
-            }
+            checkStatusCode(connection);
 
         } catch (Exception ex){
-            RCLogger.e(TAG, ex.toString());
+            handleException(ex, true);
         } finally {
             if (connection != null) {
                 connection.disconnect();
             }
         }
         RCLogger.v(TAG, "deleteBinding method ended;");
-        return result;
 
     }
 
     //-------------------------Helper methods -----------------------------//
+    private void checkStatusCode(HttpURLConnection connection) throws Exception{
+
+        try {
+            int responseCode = connection.getResponseCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+                switch (responseCode) {
+                    case 401:
+                    case 403:
+                        throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_AUTHENTICATION_FORBIDDEN);
+                    case 404:
+                        throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_HTTP_NOT_FOUND);
+                    case 408:
+                        throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_TIMED_OUT);
+                    default:
+                        throw new RCException(RCClient.ErrorCodes.ERROR_DEVICE_PUSH_NOTIFICATION_UNKNOWN_ERROR);
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Why do you need to catch and re-throw? Doesn't the `throws Exception` cover you both for IOException and RCException?

Reply:
Because we need to catch IOException;   Doesn't the throws Exception cover you both for IOException and RCException?: Yes